common/ferrostar/src/routing_adapters/osrm/algorithms.rs

```diff
@@ -0,0 +1,89 @@
+use super::models::AnyAnnotation;
+use crate::models::AnyAnnotationValue;
+use crate::routing_adapters::error::ParsingError;
+use serde_json::Value;
+use std::collections::BTreeMap;
+
+/// Get's the slice of annotations
+///
+/// Throws an [`ParsingError`] if the annotations are not present or the slice is out of bounds.
+pub(crate) fn get_annotation_slice(
+    annotations: Option<Vec<AnyAnnotationValue>>,
+    start_index: usize,
+    end_index: usize,
+) -> Result<Vec<AnyAnnotationValue>, ParsingError> {
+    annotations
+        .ok_or(ParsingError::Annotations {
+            error: "No annotations".to_string(),
```

Comment:
It feels like this function should probably either...

1. not be called at all if there no annotations (and the parameter name can change to be non-optional), OR
2. always be called but return an empty array; is having no annotations really an error?